libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1927,11 +1927,17 @@ smtutil::Expression SMTEncoder::compoundAssignment(Assignment const& _assignment
 
 	auto decl = identifierToVariable(_assignment.leftHandSide());
 
+	TypePointer commonType = Type::commonType(
+		_assignment.leftHandSide().annotation().type,
+		_assignment.rightHandSide().annotation().type
+	);
+	solAssert(commonType == _assignment.annotation().type, "");
```

Comment:
Maybe this assertion is not needed, but added it to be on a safe side.

Reply:
I'd say you could even remove this block and just use `_assignment.annotation().type` in the modified `expr` calls